tests/trainer/test_auto_trainer.py

```diff
@@ -1,43 +1,27 @@
-import numpy as np
-import torch
-from torch.utils.data import TensorDataset
+from tests.trainer.generic import std_trainer_input_1
 
-from knodle.model.logistic_regression_model import LogisticRegressionModel
 from knodle.trainer.auto_trainer import AutoTrainer
 
 
-def test_auto_train():
-    num_samples = 64
-    num_features = 16
-    num_rules = 6
-    num_classes = 2
-
-    model = LogisticRegressionModel(num_features, num_classes)
-
-    x_np = np.ones((num_samples, num_features)).astype(np.float32)
-    x_tensor = torch.from_numpy(x_np)
-    model_input_x = TensorDataset(x_tensor)
-
-    rule_matches_z = np.zeros((num_samples, num_rules))
-    rule_matches_z[0, 0] = 1
-    rule_matches_z[1:, 1] = 1
-
-    mapping_rules_labels_t = np.zeros((num_rules, num_classes))
-    mapping_rules_labels_t[:, 1] = 1
-
-    trainer = AutoTrainer(
-        name="majority",
-        model=model,
-        mapping_rules_labels_t=mapping_rules_labels_t,
-        model_input_x=model_input_x,
-        rule_matches_z=rule_matches_z,
-    )
-
-    trainer.train()
-
-    y_np = np.ones((num_samples,))
-    y_labels = TensorDataset(torch.from_numpy(y_np))
-    metrics = trainer.test(model_input_x, y_labels)
-
-    # Check whether the code ran up to here
-    assert 2 == 2
+def test_auto_train(std_trainer_input_1):
+    (
+        model,
+        model_input_x, rule_matches_z, mapping_rules_labels_t,
+        y_labels
+    ) = std_trainer_input_1
```

Comment:
Do we need a separate input for this, could not we use std_trainer_input_2 for this too?

Reply:
We already created 2 scenarios, so I thought let's keep it like it is, as it's only more thorough testing. But if you think, it'd be more consistent I could use 1 only :)